### PR TITLE
Griffin Pipeline

### PR DESCRIPTION
**Pull request by yueyaog**
This pipeline includes three workflows: GC_correction, nucleosome_profiling, and ER_status_classification. GC_correction, nucleosome_profiling WDLs were translated from the snakemake files in [Griffin](https://github.com/adoebley/Griffin/) repo. ER_status_classification was created based on the final model notebook from [Griffin_analyses](https://github.com/adoebley/Griffin_analyses) repo. All of those WDL scripts have been tested in a Terra [workspace](https://app.terra.bio/#workspaces/broadtagteam/tag1592_Griffin_MBC_ULP_WGS). The details of the Griffin Docker image could be found in a [PR](https://github.com/broadinstitute/TAG/pull/1628) in TAG repo.

### CODE REVIEW

File: Griffin/griffin_GC_correction/griffin_GC_correction.wdl [hunk 1]
```diff
@@ -0,0 +1,157 @@
...
+    input {
+        String griffin_docker
+        File bam_file
+        String sample_name
+        File mappable_regions
+        File reference_genome
+        File chrom_sizes
+        Int map_q
+        Array[Int] GC_bias_size_range
+        Int? cpu
```
Comment (yueyaog) [inline Griffin/griffin_GC_correction/griffin_GC_correction.wdl:67]: Int cpu=8

File: Griffin/griffin_GC_correction/griffin_GC_correction.wdl [hunk 2]
```diff
@@ -0,0 +1,157 @@
...
+
+        # Run griffin_GC_bias
+        conda run --no-capture-output \
+        -n griffin_env \
+        python3 /BaseImage/Griffin/scripts/griffin_GC_bias.py \
+        --bam_file_name ~{sample_name} \
+        --mappable_name ~{mappable_name} \
+        --genome_GC_frequency ./genome_GC_frequency \
+        --out_dir results \
+        --size_range ~{sep=" " GC_bias_size_range} \
```
Comment (yueyaog) [inline Griffin/griffin_GC_correction/griffin_GC_correction.wdl:140]: Extra backslash

**Review by yueyaog — COMMENTED** [review]
Some minor changes are needed

File: Griffin/griffin_nucleosome_profiling/griffin_nucleosome_profiling.wdl [hunk 3]
```diff
@@ -0,0 +1,396 @@
+version 1.0
+
+workflow nucleosome_profiling{
+    input {
+        String griffin_docker
+        File bam_file
+        File GC_bias_file
+        String sample_name
+        File reference_genome
+        File mappability_bw
```
[the remark this answers is not in the document]
Comment (lfleming1022) [reply Griffin/griffin_nucleosome_profiling/griffin_nucleosome_profiling.wdl:10]: @yueyaog And this?

File: Griffin/griffin_nucleosome_profiling/griffin_nucleosome_profiling.inputs.json [hunk 4]
```diff
@@ -0,0 +1,37 @@
...
+    "nucleosome_profiling.centromeres": "gs://gptag/resources/Griffin/Ref_hg38/hg38_centromeres.bed",
+    "nucleosome_profiling.patches": "gs://gptag/resources/Griffin/Ref_hg38/hg38_fix_patches.bed",
+    "nucleosome_profiling.size_range": [100,200],
+    "nucleosome_profiling.norm_window": [-5000,5000],
+    "nucleosome_profiling.save_window": [-1000,1000],
+    "nucleosome_profiling.center_window": [-30,30],
+    "nucleosome_profiling.fft_window": [-960,960],
+    "nucleosome_profiling.fft_index": 10,
+    "nucleosome_profiling.map_quality": 20,
+    "nucleosome_profiling.mappability_bw": "gs://gptag/resources/Griffin/Ref_hg38/k100.Umap.MultiTrackMappability.bw",
```
[the remark this answers is not in the document]
Comment (yueyaog) [reply Griffin/griffin_nucleosome_profiling/griffin_nucleosome_profiling.inputs.json:23]: Yes. mappability_bw track is needed for merge_sites task. So it is required in the inputs.json file.

**Review by lfleming1022 — APPROVED** [review]

**Review by lfleming1022 — APPROVED** [review]

**Pull request merged**